qa/integration-tests/src/test/java/io/camunda/it/rdbms/db/variables/VariableIT.java

```diff
@@ -159,14 +162,16 @@ public void shouldFindVariableWithFullFilter(final CamundaRdbmsTestApplication t
 
     assertThat(searchResult.total()).isEqualTo(1);
     assertThat(searchResult.hits()).hasSize(1);
-    assertThat(searchResult.hits().getFirst().key()).isEqualTo(randomizedVariable.key());
+    assertThat(searchResult.hits().getFirst().variableKey())
+        .isEqualTo(randomizedVariable.variableKey());
   }
 
-  private void assertVariableDbModelEqualToEntity(VariableDbModel dbModel, VariableEntity entity) {
+  private void assertVariableDbModelEqualToEntity(
+      final VariableDbModel dbModel, final VariableEntity entity) {
     assertThat(entity)
         .usingRecursiveComparison()
         .ignoringFields("bpmnProcessId", "processDefinitionId")
         .isEqualTo(dbModel);
-    assertThat(entity.bpmnProcessId()).isEqualTo(dbModel.processDefinitionId());
+    assertThat(entity.processDefinitionId()).isEqualTo(dbModel.processDefinitionId());
   }
 }
```

Comment:
🔧  Could you also include a test with the searchAfter feature? (see ProcessInstanceIT#shouldFindProcessInstanceWithSearchAfter)

Reply:
I added `VariableIT#shouldFindVariableWithSearchAfter`, but it is failing as it is returning the first page when I request the nextPage. Do you have an idea why?
Also I have noticed that, if I include `variableKey` in the sorting, I get `IndexOutOfBoundException` in https://github.com/camunda/camunda/blob/66fb7310900a460402f8d073503eb9a3583925e4/db/rdbms/src/main/java/io/camunda/db/rdbms/read/service/AbstractEntityReader.java#L105
we may need to add some validation on the searchAfter field to make sure it is consistent with the applied sorting

Reply:
I fixed it , I was missing `<include refid="io.camunda.db.rdbms.sql.Commons.keySetPageFilter"/>` in the MyBatis mapper